### PR TITLE
EmployeeBackofficeBe:  map table Department

### PR DESCRIPTION
**Pull request by LucyQuantAnalyst**
- created Department entity
- created interface
- created integration tests
-  added not nullable parametrs to SalaryDb

### CODE REVIEW

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Autowired
+    private TitleRepository titleRepository;
+
+    @Test
+    public void testLoadTitle() {
+        Integer empNo = 10001;
+        LocalDate fromDate = LocalDate.of(1986, 6, 26);
+        LocalDate toDate = LocalDate.of(9999, 1, 1);
+
```
Comment (GeorgeQuantAnalyst) [inline employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java:29]: Vytvor dalsi propemenou title = "Senior Engineer", snizi to duplicitu stejnych stringu v metode.

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/DepartmentRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        assertFalse(departmentDbWrapper.isEmpty());
+
+        DepartmentDb departmentDb = departmentDbWrapper.get();
+        assertEquals(departmentDb.getDeptNo(), "d009");
+        assertEquals(departmentDb.getDeptName(), "Customer Service");
+        }
+
+    @Test
+    public void testFindDepartments() {
```
Comment (GeorgeQuantAnalyst) [inline employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/DepartmentRepositoryIntegrationTest.java:35]: Prejmenuj metodu na testFindAllDepartmens() at je to z nazvu vice jasne ze se nacitaji vsechny.

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        Optional<TitleDb> titleDbWrapper = titleRepository.findById(new TitleDbId(empNo,"Senior Engineer", fromDate));
+        assertTrue(titleDbWrapper.isPresent());
+
+        TitleDb titleDb = titleDbWrapper.get();
+        assertEquals(titleDb.getEmpNo(), 10001);
+        assertEquals(titleDb.getTitle(), "Senior Engineer");
```
Comment (GeorgeQuantAnalyst) [inline employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java:35]: Pouzij konstantu title misto prime hodnoty.

File: employee-backoffice-be/src/main/java/com/example/employeebackofficebe/model/db/TitleDb.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+public class TitleDb {
+
+    @Id
+    @Column(name = "emp_no", nullable = false)
+    private Integer empNo;
+
+    @Id
+    @Column(name = "title", columnDefinition = "VARCHAR(50)")
+    private String title;
+    @Id
```
Comment (GeorgeQuantAnalyst) [inline employee-backoffice-be/src/main/java/com/example/employeebackofficebe/model/db/TitleDb.java:20]: Pridej prazdny radek - CleanCode

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/DepartmentRepositoryIntegrationTest.java (hunk 2 above)
Comment (LucyQuantAnalyst) [reply employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/DepartmentRepositoryIntegrationTest.java]: Hotovo

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java (hunk 1 above)
Comment (LucyQuantAnalyst) [reply employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java:30]: Hotovo

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java (hunk 3 above)
Comment (LucyQuantAnalyst) [reply employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java]: Hotovo

File: employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public void testLoadTitle() {
+        Integer empNo = 10001;
+        LocalDate fromDate = LocalDate.of(1986, 6, 26);
+        LocalDate toDate = LocalDate.of(9999, 1, 1);
+
+        Optional<TitleDb> titleDbWrapper = titleRepository.findById(new TitleDbId(empNo,"Senior Engineer", fromDate));
+        assertTrue(titleDbWrapper.isPresent());
+
+        TitleDb titleDb = titleDbWrapper.get();
+        assertEquals(titleDb.getEmpNo(), 10001);
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply employee-backoffice-be/src/test/java/com/example/employeebackofficebe/repository/TitleRepositoryIntegrationTest.java]: Hotovo

**Pull request merged**